packages/temporal/activities/mtv_controls.go

```diff
@@ -56,29 +57,43 @@ func JoinActivity(ctx context.Context, state shared.MtvRoomExposedState) error {
 	return err
 }
 
-// TODO: implement heartbeat
 func TrackTimerActivity(ctx context.Context, timerState shared.MtvRoomTimer) (shared.MtvRoomTimer, error) {
 	timerStartTime := time.Now()
 	durationBeforeTrackEnd := timerState.TotalDuration - timerState.Elapsed
-	timer := time.NewTimer(durationBeforeTrackEnd)
+	timer := time.NewTimer(durationBeforeTrackEnd) //40 sec
+	twoSecondsDuration := 2 * time.Second
 
-	select {
-	case <-timer.C:
-		// timer ended
+	for {
+		heartbeatTimer := time.NewTimer(twoSecondsDuration)
+		select {
+		case <-timer.C:
+			// timer ended
+			timerState.State = shared.MtvRoomTimerStateFinished
+			timerState.Elapsed = timerState.TotalDuration
 
-		timerState.State = shared.MtvRoomTimerStateFinished
+			return timerState, nil
 
-		return timerState, nil
+		case <-ctx.Done():
+			// context was canceled
+			cancelationTime := time.Now()
+			elapsedTimeSinceTimerStart := cancelationTime.Sub(timerStartTime)
 
-	case <-ctx.Done():
-		// context was canceled
+			timerState.State = shared.MtvRoomTimerStatePending
+			timerState.Elapsed += elapsedTimeSinceTimerStart
 
-		cancelationTime := time.Now()
-		elapsedTimeSinceTimerStart := cancelationTime.Sub(timerStartTime)
+			return timerState, nil
+		case <-heartbeatTimer.C:
 
-		timerState.State = shared.MtvRoomTimerStatePending
-		timerState.Elapsed += elapsedTimeSinceTimerStart
-
-		return timerState, nil
+			// heartbeat timer ended, going again in the loop
+			RecordHeartBeatWrapper(ctx, "status-timer-report-to-workflow")
+		}
 	}
 }
+
+func MyRecordHeartBeat(ctx context.Context, label string) {
+	activity.RecordHeartbeat(ctx, label)
+}
+
+type RecordHeartBeatWrapperType func(context.Context, string)
+
+var RecordHeartBeatWrapper RecordHeartBeatWrapperType = MyRecordHeartBeat
```

Comment:
We could simplify the code by removing `MyRecordHeartBeat` function and by assigning `activity.RecordHeartbeat` to `RecordHeartBeatWrapper` instead of `MyRecordHeartBeat`:

```suggestion
var RecordHeartBeatWrapper RecordHeartBeatWrapperType = activity.RecordHeartbeat
```